react/Header/UserAccount/UserAccount.js

```diff
@@ -46,7 +51,13 @@ export default class UserAccount extends Component {
   }
 
   handleMenuToggleClick = () => {
-    this.setState(state => ({ menuOpen: !state.menuOpen }));
+    this.setState(state => {
+      const menuState = {
+        menuOpen: !state.menuOpen
```

Comment:
Given the `onMenuToggle` event is explicitly scoped to the menu, I feel like it should be `open` instead of `menuOpen`. We probably need to have different objects for the event and the local component state.

Reply:
It makes some sense to keep these consistent, what do you think. So I should perhaps alter the internal implementation to use `open` too (assuming the context is obvious).

Reply:
To be honest, I feel like this is a risky refactor because it's creating a relationship where there shouldn't be one. What we choose to expose as an event object and the object we pass to `setState` are not logically the same thing, even if they happen to be the same right now.

Reply:
As an example of why this isn't logically correct, the event handler now has the ability to mutate the object being sent to `setState`.

Reply:
These are really good points and I hadn't considered them, folllowing a chat with Mark I agree that decoupling the state of the UserAccount from argument of the event handler is a must here. Have amended accordingly.